src/main/java/org/gridsuite/ds/server/service/parameters/impl/ParametersServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+        return getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, EVENTS_GROOVY).toString()).readAllBytes();
+    }
+
+    @Override
+    public byte[] getCurveModel() throws IOException {
+        // read the curves.groovy in the "parameters" resources
+        return getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, CURVES_GROOVY).toString()).readAllBytes();
+    }
+
+    @Override
+    public DynamicSimulationParameters getDynamicSimulationParameters(byte[] dynamicParams) throws IOException, RuntimeException {
+
+        // prepare a temp dir for current running simulation
+        Path configDir = PlatformConfig.defaultConfig().getConfigDir().orElseThrow();
+        Path workingDir = Files.createTempDirectory(configDir, WORKING_DIR_PREFIX);
+
+        // merge dynamicParams with events.par then load parametersFile in a runtime tmp directory
+        InputStream eventsIs = getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, EVENTS_PAR).toString());
+        byte[] eventsParams = eventsIs.readAllBytes();
+        if (dynamicParams.length != 0 && eventsParams.length != 0) { // suppose two models .par are valid => do merge
+            mergeParams(dynamicParams, eventsParams, workingDir, MODELS_PAR);
+        } else { // create without events par
+            Files.copy(new ByteArrayInputStream(dynamicParams), workingDir.resolve(MODELS_PAR));
+        }
+
+        // load two others files
+        for (String parFileName : List.of(NETWORK_PAR, SOLVERS_PAR)) {
+            Files.copy(getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, parFileName).toString()), workingDir.resolve(parFileName));
+        }
+
+        // load parameter file then config paths
+        DynamicSimulationParameters parameters = JsonDynamicSimulationParameters.read(getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, PARAMETERS_JSON).toString()));
+        DynaWaltzParameters dynaWaltzParameters = parameters.getExtension(DynaWaltzParameters.class);
+        dynaWaltzParameters.setParametersFile(workingDir.resolve(MODELS_PAR).toString());
+        dynaWaltzParameters.getNetwork().setParametersFile(workingDir.resolve(NETWORK_PAR).toString());
+        dynaWaltzParameters.getSolver().setParametersFile(workingDir.resolve(SOLVERS_PAR).toString());
+        return parameters;
+    }
+
+    private void mergeParams(byte[] dynamicParams, byte[] eventsParams, Path tmpDir, String modelParFileName) throws IOException {
+        InputStream dynamicParamsIs = new ByteArrayInputStream(dynamicParams);
+        InputStream eventsParamsIs = new ByteArrayInputStream(eventsParams);
+        XmlMerge xmlMerge = new SimpleXmlMerge();
+        try {
+            Document mergedDoc = xmlMerge.merge(dynamicParamsIs, eventsParamsIs);
+            xmlMerge.export(mergedDoc, Files.newOutputStream(tmpDir.resolve(modelParFileName)));
+        } catch (ParserConfigurationException | SAXException | TransformerException e) {
+            throw new RuntimeException(e);
```

Comment:
Avoid using a RuntimeException directly but prefer a subclass. We already have a lot of them in powsybl-commons.

Reply:
Done